packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -230,14 +230,25 @@ const Separator = ({ separator }: Pick<BreadcrumbsProps, 'separator'>) => (
 const BreadcrumbsItem = React.forwardRef((props, forwardedRef) => {
   const { children: childrenProp, className, ...rest } = props;
 
-  const defaultAs = !!props.href ? Anchor : !!props.onClick ? 'button' : 'span';
+  const resolvedAs = !!props.href ? 'a' : !!props.onClick ? 'button' : 'span';
+
+  const resolvedAsComponentsMap: Record<
+    typeof resolvedAs,
+    PolymorphicForwardRefComponent<'a' | 'button'> | 'span'
+  > = {
+    a: Anchor,
+    button: Button,
```

Comment:
This implementation looks overcomplicated. Why not just use `<Button styleType="borderless">` (instead of `<Box>`) and keep everything else the same as before?

Reply:
The reason was that with using a `Button`, we get the `Button`'s hover background when hovering over breadcrumb items that are supposed to be anchors.

<details>
<summary>Screen-recordings</summary>

HTML Demo Page:

https://github.com/iTwin/iTwinUI/assets/45748283/a5fbec2b-2933-4348-8daa-7def7db109e9

`Breadcrumbs → Links` story when using `Button styleType="borderless">`:

https://github.com/iTwin/iTwinUI/assets/45748283/44808e1e-efea-4c2b-bd5d-4e0fe082806f

</details>

<details>
<summary>Diff</summary>

```diff
diff --git a/packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx b/packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx
index 4c39eecdd..fe7caa668 100644
--- a/packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx
+++ b/packages/itwinui-react/src/core/Breadcrumbs/Breadcrumbs.tsx
@@ -230,30 +230,20 @@ const Separator = ({ separator }: Pick<BreadcrumbsProps, 'separator'>) => (
 const BreadcrumbsItem = React.forwardRef((props, forwardedRef) => {
   const { children: childrenProp, className, ...rest } = props;
 
-  const resolvedAs = !!props.href ? 'a' : !!props.onClick ? 'button' : 'span';
-
-  const resolvedAsComponentsMap: Record<
-    typeof resolvedAs,
-    PolymorphicForwardRefComponent<'a' | 'button'> | 'span'
-  > = {
-    a: Anchor,
-    button: Button,
-    span: 'span',
-  };
-
+  const defaultAs = !!props.href ? Anchor : !!props.onClick ? 'button' : 'span';
   const children =
-    resolvedAs === 'button' ? <span>{childrenProp}</span> : childrenProp;
+    defaultAs === 'button' ? <span>{childrenProp}</span> : childrenProp;
 
   return (
-    <Box
-      as={resolvedAsComponentsMap[resolvedAs] as 'a'}
+    <Button
+      as={defaultAs as 'a'}
       className={cx('iui-breadcrumbs-content', className)}
-      data-iui-variant={resolvedAs === 'button' ? 'borderless' : undefined}
+      styleType='borderless'
       ref={forwardedRef}
       {...rest}
     >
       {children}
-    </Box>
+    </Button>
   );
 }) as PolymorphicForwardRefComponent<'a'>;
 BreadcrumbsItem.displayName = 'Breadcrumbs.Item';
```

</details>

Reply:
> we get the Button's hover background when hovering over breadcrumb items that are supposed to be anchors.

Is this an actual problem? The breadcrumbs are still underlined to indicate that they're anchors rather than buttons, and the additional hover background is a more accurate indicator of the true size of the breadcrumb.

@FlyersPh9

Reply:
I just tried this in the CSS demo page like so:
```diff
<a
-  class='iui-anchor iui-breadcrumbs-content'
+  class='iui-anchor iui-button-base iui-button iui-field iui-breadcrumbs-content'
+  data-iui-variant='borderless'
  href='#'>
  Root
</a>
```

And am getting this result:
![Screenshot 2024-04-25 at 2 18 32 PM](https://github.com/iTwin/iTwinUI/assets/849817/870ff18c-1430-4c3b-a635-d7c8a063d95b)

Reply:
I think you need a `<span>` around the text to make it truncate.

If you use `Button`, this should be handled automatically.

Reply:
Jon and I tried using a `Button`. Although it added a span, that was not sufficient to solve the issue.

I was able to fix the issue by adding some custom styles to the span.

<details>
<summary>Screenshot</summary>

![image](https://github.com/iTwin/iTwinUI/assets/45748283/6f04d9eb-72fe-4936-b1c1-8421303bed2b)

</details>

We can try investigating in which selector to add those styles or if there is a better way to fix the issue. But since making the BreadcrumbsItem look like a Button is not directly related to this iui-field PR, I think it can be done later. If so, we can add an after PR TODO.

For now, I simplified the code a bit more (https://github.com/iTwin/iTwinUI/pull/1919/commits/1ddac82299ab427d48ef97abbea0a175fd57ccf4).

Reply:
I think this diffs should be enough to fix the truncation?

<details><summary>breadcrumbs.scss</summary>

```diff
@@ -35,6 +35,13 @@
 .iui-breadcrumbs-content {
   padding-inline: var(--iui-size-xs);
   overflow: hidden;
   text-overflow: ellipsis;
   line-height: var(--iui-component-height);
 
+  &,
+  > * {
+    overflow: hidden;
+    white-space: nowrap;
+    text-overflow: ellipsis;
+  }
+
   &:where(a) {
     &:focus {
       outline-offset: -1px;
@@ -54,12 +61,6 @@
   &:where(button) {
     @include base.iui-button;
@@ -61,5 +61,5 @@
     &[aria-current] {
       color: var(--iui-color-text);
     }
-
-    > * {
-      overflow: hidden;
-      white-space: nowrap;
-      text-overflow: ellipsis;
-    }
```

</details>

Reply:
Yes, Jon and I had tried that (except we didn't add the `&` in the selector) before and it had worked. But I didn't confirm if anything else breaks as a result. Since it wasn't related to this PR, we considered having it as an after PR TODO.

@FlyersPh9 If you feel that nothing breaks on the css side with this change, I can make this change on the react side following your css/html demo changes.

Reply:
Got it working in https://github.com/iTwin/iTwinUI/pull/1919/commits/3c56408b945a2c7faf5e923453c5ab6797fc58cf.

Reply:
Great, made the change in React too (https://github.com/iTwin/iTwinUI/pull/1919/commits/d2d6e369b94ca470760d89da6d59bce2c54f7a24).